components/TextFieldWithCopyPaste.tsx

```diff
@@ -0,0 +1,115 @@
+import ClearIcon from '@mui/icons-material/Clear';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+import ContentPasteGoIcon from '@mui/icons-material/ContentPasteGo';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import React, { useState } from 'react';
+
+import useSupportsClipboardRead from '../hooks/useSupportsClipboardRead';
+import Toast, { ToastProps } from './Toast';
+
+type Props = {
```

Comment:
Since this is a generic component, lets make this an interface and export it for later use elsewhere.